src/render.rs

```diff
@@ -0,0 +1,263 @@
+//! Rendering is RLBot's ability to draw directly inside the game window.
+
+use flat;
+use flatbuffers::{FlatBufferBuilder, WIPOffset};
+use rlbot::RLBot;
+use std::error::Error;
+
+/// A render group in the process of being built.
+///
+/// The drawing methods in this class simply collect a list of items to be
+/// drawn later. Nothing will actually be drawn to screen until
+/// [`render`](RenderGroup::render) is called.
+///
+/// # Examples
+///
+/// ## Basic rendering
+///
+/// ```no_run
+/// # use rlbot::flat::ColorArgs;
+/// # use std::error::Error;
+/// #
+/// # fn main() -> Result<(), Box<Error>> {
+/// let rlbot = rlbot::init()?;
+/// let mut group = rlbot.begin_render_group(1234);
+/// let yellow = group.color_rgb(255, 255, 0);
+/// group.draw_string_2d((10.0, 10.0), (2, 2), "I am text!", yellow);
+/// group.render()?;
+/// # Ok(())
+/// # }
+/// ```
+///
+/// ## Clearing the screen
+///
+/// ```no_run
+/// # use std::error::Error;
+/// #
+/// # fn main() -> Result<(), Box<Error>> {
+/// let rlbot = rlbot::init()?;
+/// let group = rlbot.begin_render_group(1234);
+/// group.render()?;
+/// # Ok(())
+/// # }
+/// ```
+pub struct RenderGroup<'a> {
+    rlbot: &'a RLBot,
+    id: i32,
+    builder: FlatBufferBuilder<'a>,
+    messages: Vec<WIPOffset<flat::RenderMessage<'a>>>,
+}
+
+impl<'a> RenderGroup<'a> {
+    pub(crate) fn new(rlbot: &'a RLBot, id: i32) -> RenderGroup<'a> {
+        RenderGroup {
+            rlbot,
+            id,
+            builder: FlatBufferBuilder::new_with_capacity(1024),
+            messages: Vec::new(),
+        }
+    }
+}
+
+/// A color that can be used to draw in a [`RenderGroup`].
+#[derive(Copy, Clone)]
+pub struct Color<'a>(WIPOffset<flat::Color<'a>>);
+
+impl<'a> RenderGroup<'a> {
+    /// Send the collected drawings to RLBot to be rendered to screen.
+    pub fn render(mut self) -> Result<(), Box<Error>> {
+        let messages = self.builder.create_vector(&self.messages);
+
+        let render_group = {
+            let mut rg = flat::RenderGroupBuilder::new(&mut self.builder);
+            rg.add_renderMessages(messages);
+            rg.add_id(self.id);
+            rg.finish()
+        };
+
+        self.builder.finish(render_group, None);
+        let data = self.builder.finished_data();
+        self.rlbot.render_group(data)?;
+        Ok(())
+    }
+
+    /// Create a color with the given **a**lpha, **r**ed, **g**reen, and
+    /// **b**lue. An alpha of 255 is fully opaque, and 0 is fully transparent.
+    ///
+    /// A color can only be used with the `RenderGroup` that created it.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use rlbot::RenderGroup;
+    /// # let mut group: RenderGroup = unsafe { ::std::mem::uninitialized() };
+    /// let transbluecent = group.color_argb(127, 0, 0, 255);
+    /// ```
+    pub fn color_argb(&mut self, a: u8, r: u8, g: u8, b: u8) -> Color<'a> {
```

Comment:
This order of arguments matches the order it's defined in, but personally I find rgba more intuitive. This is subjective though, so feel free to ignore this.

Reply:
Thanks for reviewing this. Like you said this is subjective, and in cases like this where there's not a clear reason for changing I think it's better to stick with the defined order, so I think I'll keep it as is.